docs/usage/annotated_types.md

```diff
@@ -0,0 +1,102 @@
+---
+description: Support for annotated types including NamedTuple and TypedDict.
```

Comment:
"annotated types" has a particular meaning - https://github.com/annotated-types/annotated-types so probably not a good name.

Perhaps "types with fields", and just start by saying there are four supported types with fields:
* base models
* dataclasses
* typed dicts
* named tuples

Reply:
Should this entire page be renamed then? The textual fix is simple

Reply:
Yes I think so.